pkg/utils/controller/controller.go

```diff
@@ -86,6 +86,24 @@ func NewUserError(err error) error {
 	return nil
 }
 
+// NewCreateIgnoreAlreadyExistError returns ErrExpectedBehavior type error if the error is already exist.
+// Otherwise, returns ErrAPIServerError type error.
+func NewCreateIgnoreAlreadyExistError(err error) error {
+	if !apierrors.IsAlreadyExists(err) {
+		return NewAPIServerError(false, err)
+	}
+	return NewExpectedBehaviorError(err)
+}
+
+// NewUpdateIgnoreConflictError returns ErrExpectedBehavior type error if the error is conflict.
+// Otherwise, returns ErrAPIServerError type error.
+func NewUpdateIgnoreConflictError(err error) error {
+	if !apierrors.IsConflict(err) {
+		return NewAPIServerError(false, err)
+	}
+	return NewExpectedBehaviorError(err)
+}
```

Comment:
make it generic?

for example,
```
type IsExpectedErrorFn func(error) bool

// NewAPIServerError returns error types when accessing data from cache or API server.
func NewAPIServerOrExpectedBehaviorError(fromCache bool, err error, f IsExpectedErrorFn) error {
	if err != nil {
		if f(err) {
			return NewExpectedBehaviorError(err)
		}

		if fromCache && isUnexpectedCacheError(err) {
			return NewUnexpectedBehaviorError(err)
		}
		return fmt.Errorf("%w: %v", ErrAPIServerError, err.Error())
	}
	return nil
}
```

You can call it using NewAPIServerOrExpectedBehaviorError(false, err, apierrors.IsConflict)

Reply:
SG except those are only for write operations in my mind

Reply:
btw, these two are the only cases I can think of now. If that's the case, does it still make sense to have a generic one?

Reply:
I'm fine and we could do the refactor later.